js/widgets.js

```diff
@@ -26,11 +87,30 @@ const baseURL = `${parser.protocol}//${parser.host}`
  */
 function chunk (array, n) {
   return Array
-    .apply(null, Array(Math.ceil(array.length / n))).map((x, i) => i) // range
+    .apply(null, Array(Math.ceil(array.length / n)))
+    .map((x, i) => i)
     .map((x, i) => array.slice(i * n, i * n + n))
 }
 
 /**
+ * Return a function, that, as long as it continues to be invoked,
+ * will not be triggered. The function will be called after
+ * it stops being called for `wait` milliseconds. Inspired by:
+ * https://davidwalsh.name/javascript-debounce-function
+ */
+function debounce (func, wait = 350) {
```

Comment:
We have a `debounce` dependency (I can't say if it's still in use or not). It might be interesting to either drop the useless dependency and use this piece everywhere (or reuse the dependency)